packages/icons/src/components/Icon.tsx

```diff
@@ -18,6 +18,7 @@ import { IconName, icons } from '../icons';
 type IconProps = {
     name: IconName;
     size?: IconSize;
+    customSize?: number;
```

Comment:
You should never have `customSize` and `size` at same time. Please use this util https://github.com/sindresorhus/type-fest/blob/main/source/require-exactly-one.d.ts